### PR TITLE
Implements dropdown for filter selection

### PR DESCRIPTION
**Pull request by anishaswain**
https://user-images.githubusercontent.com/16955978/117258507-adc3f880-ae6a-11eb-8131-1869609bfe84.mov

### CODE REVIEW

File: src/components/TableFilterSelection/index.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+  Dropdown,
+  DropdownToggle,
+  DropdownItem,
+  DropdownPosition,
+} from '@patternfly/react-core';
 import Select from '@/components/Select';
-
-import Button from '../Button';
+import styles from './index.less';
+// import Button from '../Button';
```
Comment (gurbirkalsi) [inline src/components/TableFilterSelection/index.js]: Do we need this comment here?

**Review by gurbirkalsi — CHANGES_REQUESTED** [review]
Great start @anishaswain! A few design considerations & bugs we'll need to address before merging: 
- Hovering over the toggled filter selection component highlights every field name in red
Screen Shot 2021-05-06 at 4 59 46 PM

- Closing a selected filter from a field's dropdown menu doesn't remove the selected filter
- The width of the dropdown menu for a field scales according to the page width. Can we consider setting `width: auto` to set the width for the menu? 

Screen Shot 2021-05-06 at 5 10 14 PM

- Clicking the "Save" button doesn't close the toggled filter selection component
- We can utilize the [Toolbar](https://www.patternfly.org/v4/components/toolbar) component to align the "Apply Filters" and "Clear Filters" buttons with "Compare Iterations"

Screen Shot 2021-05-06 at 5 09 21 PM

File: src/components/TableFilterSelection/index.js (hunk 1 above)
Comment (anishaswain) [reply src/components/TableFilterSelection/index.js]: Removed

**Comment by anishaswain** [comment]
> Great start @anishaswain! A few design considerations & bugs we'll need to address before merging:
> 
> * Hovering over the toggled filter selection component highlights every field name in red
> * The width of the dropdown menu for a field scales according to the page width. Can we consider setting `width: auto` to set the width for the menu?
> * Clicking the "Save" button doesn't close the toggled filter selection component
> * We can utilize the [Toolbar](https://www.patternfly.org/v4/components/toolbar) component to align the "Apply Filters" and "Clear Filters" buttons with "Compare Iterations"

FIXED the above points and added `ToolBar` component

> * Closing a selected filter from a field's dropdown menu doesn't remove the selected filter

@gurbirkalsi this feature is not functional in the `main` branch itself. I haved opened up another PR #164 which will fix this bug

**Review by gurbirkalsi — DISMISSED** [review]

**Comment by aquibbaig** [comment]
@anishaswain can you check why the dropdown floats to the left in my case?
Screenshot

**Comment by anishaswain** [comment]
> @anishaswain can you check why the dropdown floats to the left in my case?

@aquibbaig could you please help me with your screen resolution? I am not being able to reproduce the issue.

It is a Mac just like yours, 2560 x 1600 probably

**Comment by aquibbaig** [comment]
this issue has now resolved @anishaswain, it looks great

File: src/components/TableFilterSelection/index.js [hunk 2]
```diff
@@ -51,49 +57,66 @@ export default class TableFilterSelection extends Component {
...
+              Save
+            </Button>
+            {/* </div> */}
+          </Form>
+        </div>
+      </DropdownItem>,
+    ];
+
+    return (
+      <div id="dropdownDiv">
```
Comment (aquibbaig) [inline src/components/TableFilterSelection/index.js]: are we using this id somewhere?

**Review by aquibbaig — DISMISSED** [review]

File: src/components/TableFilterSelection/index.js (hunk 2 above)
Comment (anishaswain) [reply src/components/TableFilterSelection/index.js]: fixed!

File: src/components/TableFilterSelection/index.js [hunk 3]
```diff
@@ -51,49 +57,66 @@ export default class TableFilterSelection extends Component {
...
+            </Flex>
+            <Button
+              variant="primary"
+              type="submit"
+              isDisabled={updateFiltersDisabled}
+              onClick={this.onFilterTable}
+            >
+              Save
+            </Button>
+            {/* </div> */}
```
Comment (dbutenhof) [inline src/components/TableFilterSelection/index.js]: Is the commented `</div>` here inside empty brackets important somehow?

**Review by dbutenhof — DISMISSED** [review]
So far as I can tell, this looks OK. Compared to the early screen shots, combining the filter and compare buttons into a toolbar looks like an improvement.

One minor comment/question ...

File: src/components/TableFilterSelection/index.js (hunk 3 above)
Comment (anishaswain) [reply src/components/TableFilterSelection/index.js]: Node, left it by by mistake. Resolved now.

**Review by aquibbaig — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]

**Comment by aquibbaig** [comment]
@anishaswain any updates on this PR? I think it's ready for a merge

**Comment by anishaswain** [comment]
> @anishaswain any updates on this PR? I think it's ready for a merge

@aquibbaig yes, we should merge it.

**Pull request merged**